File: app/src/main/java/com/agarwal/arpit/androidhub/lifecycle/ActivityA.java
package com.agarwal.arpit.androidhub.lifecycle;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;

import com.agarwal.arpit.androidhub.R;
import com.agarwal.arpit.androidhub.Utils.Utils;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import timber.log.Timber;

public class ActivityA extends AppCompatActivity{

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_a);
        Timber.v("onCreate");

        Intent intent = new Intent(this,ActivityB.class);
        startActivityForResult(intent,0);

    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
        Timber.v("onActivityResult");
        if (true){
            if (requestCode== 0 && resultCode == Activity.RESULT_OK){
                String val = String.valueOf(data.getExtras().getString("result"));
                Utils.showToast(this,val);
            }else if (resultCode == Activity.RESULT_CANCELED){
                Utils.showToast(this,"No result returned for the activity with request code");
            }
        }
    }

    @Override
    protected void onStart() {
        super.onStart();
        Timber.v("onStart");
    }

    // This callback is called only when there is a saved instance that is previously saved by using
    // onSaveInstanceState(). We restore some state in onCreate(), while we can optionally restore
    // other state here, possibly usable after onStart() has completed.
    // The savedInstanceState Bundle is same as the one used in onCreate().
    @Override
    protected void onRestoreInstanceState(Bundle savedInstanceState) {
        super.onRestoreInstanceState(savedInstanceState);
        Timber.v("onRestoreInstanceState");

    }

    @Override
    protected void onPostCreate(@Nullable Bundle savedInstanceState) {
        super.onPostCreate(savedInstanceState);
        Timber.v("onPostCreate");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Timber.v("onResume");

    }

    @Override
    protected void onPostResume() {
        super.onPostResume();
        Timber.v("onPostResume");
    }

    @Override
    public void onAttachedToWindow() {
        super.onAttachedToWindow();
        Timber.v("onAttachedToWindow");

    }


    @Override
    protected void onPause() {
        super.onPause();
        Timber.v("onPause");


    }

    @Override
    protected void onStop() {
        super.onStop();
        Timber.v("onStop");

    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        Timber.v("onSaveInstanceState");

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Timber.v("onDestroy");

    }
}

File: app/src/main/java/com/agarwal/arpit/androidhub/lifecycle/ActivityA.kt
package com.agarwal.arpit.androidhub.lifecycle


import android.app.Activity
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.agarwal.arpit.androidhub.R
import com.agarwal.arpit.common.logging.HubLog
import com.agarwal.arpit.common.utils.showToast
import kotlinx.android.synthetic.main.activity_a.*

class ActivityA : AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_a)

        HubLog.v("onCreate")


        activitya_btn.setOnClickListener {
            val intent = Intent(this, ActivityA::class.java)
            startActivity(intent)
        }

        activityb_btn.setOnClickListener {
            val intent = Intent(this, ActivityB::class.java)
            startActivityForResult(intent, 0)
        }

        activityc_btn.setOnClickListener {
            val intent = Intent(this, ActivityC::class.java)
            startActivity(intent)
        }
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        HubLog.v("onActivityResult")
        if (true) {
            if (requestCode == 0 && resultCode == Activity.RESULT_OK) {
                val `val` = data!!.extras!!.getString("result").toString()
                showToast(`val`)
            } else if (resultCode == Activity.RESULT_CANCELED) {
                showToast("No result returned for the activity with request code")
            }
        }
    }

    override fun onStart() {
        super.onStart()
        HubLog.v("onStart")
    }

    // This callback is called only when there is a saved instance that is previously saved by using
    // onSaveInstanceState(). We restore some state in onCreate(), while we can optionally restore
    // other state here, possibly usable after onStart() has completed.
    // The savedInstanceState Bundle is same as the one used in onCreate().
    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        super.onRestoreInstanceState(savedInstanceState)
        HubLog.v("onRestoreInstanceState")

    }

    override fun onPostCreate(savedInstanceState: Bundle?) {
        super.onPostCreate(savedInstanceState)
        HubLog.v("onPostCreate")
    }

    override fun onResume() {
        super.onResume()
        HubLog.v("onResume")

    }

    override fun onPostResume() {
        super.onPostResume()
        HubLog.v("onPostResume")
    }

    override fun onAttachedToWindow() {
        super.onAttachedToWindow()
        HubLog.v("onAttachedToWindow")

    }


    override fun onPause() {
        super.onPause()
        HubLog.v("onPause")


    }

    override fun onStop() {
        super.onStop()
        HubLog.v("onStop")

    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        HubLog.v("onSaveInstanceState")

    }

    override fun onDestroy() {
        super.onDestroy()
        HubLog.v("onDestroy")

    }
}
